src/quantity_api/__init__.py

```diff
@@ -1,14 +1,29 @@
 """Quantity API."""
 
-from typing import Any, Final, Protocol, Self, runtime_checkable
+from typing import Any, Final, Protocol, Self, runtime_checkable, Optional
 
 import optype as op
 
 type Unit = Any  # TODO: unit-api
+type Dimension = Any  # TODO: dimension-api
 
 __version__: Final = "0.0.1.dev0"
 __all__ = ["__version__", "Quantity"]
 
+
+@runtime_checkable
+class QuantityNamespace[Q, U: Unit, D: Dimension](Protocol):
```

Comment:
If we like `__metrology_namespace__` then 

```suggestion
class MetrologyNamespace[Q, U: Unit, D: Dimension](Protocol):
```